sentinelhub/sentinelhub_batch.py

```diff
@@ -0,0 +1,398 @@
+        """ Updates information about a batch request
+
+        :return: Batch request info
+        :rtype: dict
+        """
+        url = self._get_process_url(self.config, request_id=self.request_id)
+        self._request_info = get_auth_json(url)
+
+    @property
+    def evalscript(self):
+        """ Provides an evalscript used by a batch request
+
+        :return: An evalscript
+        :rtype: str
+        """
+        return self.info['processRequest']['evalscript']
+
+    @property
+    def bbox(self):
+        """Provides a bounding box used by a batch request
+
+        :return: An area bounding box together with CRS
+        :rtype: BBox
+        """
+        bbox, _, crs = self._parse_bounds_payload()
+        return BBox(bbox, crs)
+
+    @property
+    def geometry(self):
+        """ Provides a geometry used by a batch request
+
+        :return: An area geometry together with CRS
+        :rtype: Geometry
+        """
+        _, geometry, crs = self._parse_bounds_payload()
+        if geometry is None:
+            geometry = self.bbox.geometry
+        return Geometry(geometry, crs)
+
+    @staticmethod
+    def iter_requests(search=None, sort=None, user_id=None, config=None, **kwargs):
+        """ Iterate existing batch requests
+
+        :param search: Filter requests by a search query
+        :type search: str or None
+        :param sort: Sort obtained batch requests in a specific order
+        :type sort: str or None
+        :param user_id: Filter requests by a user id who defined a request
+        :type user_id: str or None
+        :param config: A configuration object
+        :type config: SHConfig or None
+        :param kwargs: Any additional parameters to include in a request query
+        :return: An iterator over existing batch requests
+        :rtype: Iterator[SentinelHubBatch]
+        """
+        url = SentinelHubBatch._get_process_url(config)
+        params = _remove_undefined_params({
+            'search': search,
+            'sort': sort,
+            'userid': user_id,
+            **kwargs
+        })
+        for request_info in _iter_pages(url, **params):
+            yield SentinelHubBatch(request_info=request_info, config=config)
+
+    @staticmethod
+    def get_latest_request(config=None):
+        """ Provides a batch request that has been created the latest
+        """
+        # This should be improved once sort parameter will be supported
+        batch_requests = list(SentinelHubBatch.iter_requests(config=config))
+        return max(*batch_requests, key=lambda request: request.info['created'])
+
+    def delete(self):
+        """ Delete a batch job request
+        """
+        url = self._get_process_url(self.config, request_id=self.request_id)
+        return get_auth_json(url, request_type=RequestType.DELETE)
+
+    def start_analysis(self):
+        """ Starts analysis of a batch job request
+        """
+        return self._call_job('analyse')
```

Comment:
These job names should be enums.

Reply:
Since each of these strings is used only in a single location I don't think it is justified to have an enum for this. In this case it would only bring some boilerplate code with it.